src/shipit_static_analysis/default.nix

```diff
@@ -0,0 +1,171 @@
+{ releng_pkgs 
+}: 
+
+let
+
+  inherit (releng_pkgs.lib) mkPython fromRequirementsFile filterSource;
+  inherit (releng_pkgs.pkgs) writeScript makeWrapper fetchurl ;
+  inherit (releng_pkgs.pkgs.stdenv) mkDerivation;
+  inherit (releng_pkgs.pkgs.lib) fileContents optional licenses;
+  inherit (releng_pkgs.tools) pypi2nix;
+
+  inherit (releng_pkgs.pkgs) rustStable cacert fetchFromGitHub pythonFull which autoconf213
+    perl unzip zip gnumake yasm pkgconfig xlibs gnome2 pango dbus dbus_glib
+    alsaLib libpulseaudio gstreamer gst_plugins_base gtk3 glib freetype fontconfig cairo
+    gobjectIntrospection git mercurial openssl cmake  python27 gawk binutils
+    gcc gcc-unwrapped glibc clang clang-tools llvmPackages_39 libstdcxx5;
+
+  inherit (releng_pkgs.pkgs.pythonPackages) setuptools;
+
+  python = import ./requirements.nix { inherit (releng_pkgs) pkgs; };
+  name = "mozilla-shipit-static-analysis";
+  dirname = "shipit_static_analysis";
+
+  robustcheckout = mkDerivation {
+    name = "robustcheckout";
+    src = fetchurl { 
+      url = "https://hg.mozilla.org/hgcustom/version-control-tools/archive/1a8415be17e8.tar.bz2";
+      sha256 = "005n7ar8cn7162s1qx970x1aabv263zp7mxm38byxc23nzym37kn";
+    };
+    installPhase = ''
+      mkdir -p $out
+      cp -rf hgext/robustcheckout $out
+    '';
+    doCheck = false;
+    buildInputs = [];
+    propagatedBuildInputs = [ ];
+    meta = {
+      homepage = "https://hg.mozilla.org/hgcustom/version-control-tools";
+      license = licenses.mit;
+      description = "Mozilla Version Control Tools: robustcheckout";
+    };
+  };
+
+  mercurial' = mercurial.overrideDerivation (old: {
+    postInstall = old.postInstall + ''
+      cat > $out/etc/mercurial/hgrc <<EOF
+[web]
+cacerts = ${cacert}/etc/ssl/certs/ca-bundle.crt
+
+[extensions]
+purge =
+robustcheckout = ${robustcheckout}/robustcheckout/__init__.py
+EOF
+    '';
+  });
+
+  self = mkPython {
+    inherit python name dirname;
+    inProduction = true;
+    version = fileContents ./../../VERSION;
+    src = filterSource ./. { inherit name; };
+    buildInputs =
+      fromRequirementsFile ./requirements-dev.txt python.packages;
+    propagatedBuildInputs =
+      fromRequirementsFile ./requirements.txt python.packages
+      ++ [
+        # Needed for the static analysis
+        clang
+        clang-tools
+        libstdcxx5
+        gcc
+        gcc-unwrapped
+        glibc
+        gawk
+
+        # TODO: Use nixpkgs-mozilla gecko buildInputs
```

Comment:
@garbas I'm not sure now that we can use directly the nixpkgs-mozilla as we need these in propagatedBuildInputs so that `mach configure / build` can be run inside the docker container (and not built before).

So i used the dependencies and tweaked them to match my need.
`mach configure` runs fine in development environment, but i have some trouble making it run in docker (failure on moz.configure/old.configure)